composer/utils/checkpoint.py

```diff
@@ -621,7 +621,13 @@ def load_sharded_checkpoint(
             # 1. Load metadata first for backwards compatability check
             # We need to check if the "optimizers" is at the root of the state dict to determine
             # how to load the optimizer state.
-            metadata = storage_reader.read_metadata()
+            try:
+                metadata = storage_reader.read_metadata()
+            except AttributeError:
```

Comment:
should we catch the more specific error here? instead of any attribute error?

Reply:
addressed